src/components/editor/inputs/InputLookupValue.jsx

```diff
@@ -0,0 +1,220 @@
+              className="btn btn-link py-0 pe-0"
+              onClick={handleOwnUriClick}
+            >
+              Enter your own URI and label
+            </button>
+          </div>
+          <div className="input-group">
+            <input
+              type="text"
+              className={controlClasses.join(" ")}
+              placeholder={`Enter lookup query for ${propertyTemplate.label}`}
+              aria-label={`Enter lookup query for ${propertyTemplate.label}`}
+              onChange={handleChangeDiacritics}
+              onKeyDown={handleKeyDown}
+              onBlur={handleBlurDiacritics}
+              onClick={handleClickDiacritics}
+              value={currentContent}
+              ref={inputRef}
+              id={id}
+            />
+            <button
+              className="btn btn-outline-secondary"
+              type="button"
+              onClick={handleLookupClick}
+              aria-label={`Submit lookup`}
+              data-testid={`Submit lookup`}
+            >
+              <FontAwesomeIcon icon={faSearch} />
+            </button>
+            <div className="invalid-feedback">{value.errors.join(", ")}</div>
+          </div>
+        </div>
+        <div className="col-md-auto d-flex align-items-end">
+          <button
+            className="btn btn-link fs-4 py-0"
+            id={diacriticsBtnId}
+            aria-label={`Select diacritics for ${currentContent}`}
+            data-testid={`Select diacritics for ${currentContent}`}
+            onClick={toggleDiacritics}
+            onBlur={handleBlurDiacritics}
+          >
+            &auml;
+          </button>
+        </div>
+        <div className="col-sm-1 d-flex align-items-end">
+          <button
+            type="button"
+            className="btn btn-sm"
+            aria-label={`Remove ${currentContent}`}
+            data-testid={`Remove ${currentContent}`}
+            onClick={handleRemoveClick}
+          >
+            <FontAwesomeIcon className="trash-icon" icon={faTrashAlt} />
+          </button>
+        </div>
+      </div>
+      <div className="row">
+        <div className="col">
+          <div className="form-text mt-0 mb-2">
+            Lookup with: {authorityLabels.join(", ")}
+          </div>
+        </div>
+      </div>
+      <div className="row">
+        <DiacriticsSelection
+          id={diacriticsId}
+          handleAddCharacter={handleAddCharacter}
+          closeDiacritics={closeDiacritics}
+          showDiacritics={showDiacritics}
+        />
+      </div>
+      <ResourceList property={value.property} />
+      <div className="row">
+        <Lookup
+          property={value.property}
+          propertyTemplate={propertyTemplate}
+          show={showLookup}
+          hideLookup={hideLookup}
+          handleUpdateLiteral={handleUpdateLiteral}
+          handleUpdateURI={handleUpdateURI}
+          query={query}
+        />
+      </div>
+    </React.Fragment>
```

Comment:
this is long.  I assume you chose to leave it this way rather than breaking it up into pieces?  Not sure what's better but maybe well named pieces would be helpful?

Reply:
I agree. Let me componentize some of this.